src/backend/metal/src/device.rs

```diff
@@ -359,7 +359,11 @@ impl PhysicalDevice {
                 shared_textures: !os_is_mac,
                 base_instance: Self::supports_any(&device, BASE_INSTANCE_SUPPORT),
                 dual_source_blending: Self::supports_any(&device, DUAL_SOURCE_BLEND_SUPPORT),
-                format_depth24_stencil8: device.d24_s8_supported(),
+                format_depth24_stencil8: if os_is_mac {
+                    device.d24_s8_supported()
```

Comment:
Maybe we could simply say `format_depth24_stencil8: os_is_mac && device.d24_s8_supported(),`?